src/integrators/integrator_utils.jl

```diff
@@ -186,6 +186,28 @@ function step_reject_controller!(integrator,alg::JVODE)
   integrator.dt *= integrator.qold
 end
 
+function stepsize_controller!(integrator, alg::QNDF)
+  cnt = integrator.iter
+  if cnt <= 3
+    # call std controller
+    qtmp = integrator.EEst^(1/(get_current_adaptive_order(integrator.alg,integrator.cache)+1))/integrator.opts.gamma
+    @fastmath q = max(inv(integrator.opts.qmax),min(inv(integrator.opts.qmin),qtmp))
+    integrator.qold = integrator.dt/q
+    return q
+  else
+    q = integrator.dt/integrator.cache.h
```

Comment:
what's this?

Reply:
`h (next_step_size) = (1/q) * integrator.dt` and we know `h` from `stepsize_and_order!` function.

Reply:
oh okay. Precalculated. Didn't see that.